src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2841,6 +2841,50 @@ exprt c_typecheck_baset::do_special_functions(
     overflow.type() = bool_typet{};
     return overflow;
   }
+  else if(identifier == CPROVER_PREFIX "enum_is_in_range")
+  {
+    // Check correct number of arguments
+    if(expr.arguments().size() != 1)
+    {
+      error().source_location = expr.source_location();
+      error() << "__enum_is_in_range expects one argument" << eom;
+      std::ostringstream error_message;
+      error_message << expr.source_location().as_string() << ": "
+                    << "__enum_is_in_range expects one argument, "
+                    << expr.arguments().size() << " were provided";
+      throw invalid_source_file_exceptiont{error_message.str()};
```

Comment:
I believe that in this code we would just use `throw 0;`

Reply:
Fixed in both places.

Reply:
@tautschnig we _do_ do that, but just as a historical artifact. We'd agreed in the past we wouldn't do that for new code anymore.